### PR TITLE
Add generator-jhipster as an npm/yarn dependency if the app is a microservice

### LINKED ISSUE #4218: Add generator-jhipster as an npm dependency if the app is a microservice
See #4217 

We should ensure that the same version of jhipster is used when regenererating a micro-service.

### PR DESCRIPTION
**Pull request by PierreBesson**
Fix #4218
Add a minimal package.json and npm/yarn install step after generation for skipClient apps. 

TODO
- [x] more testing
- [x] a flag to disable the feature
- [x] checking that it works with the additional module installation question

### CODE REVIEW

File: generators/app/index.js [hunk 1]
```diff
@@ -336,6 +337,29 @@ module.exports = JhipsterGenerator.extend({
...
+                var done = this.async();
+                var errorLocalJhipsterInstall = function (err) {
+                    if (err) {
+                        this.warning('Could not install generator-jhipster locally.');
+                    }
+                    done();
+                };
+                if(this.yarnInstall) {
+                    this.log(chalk.bold('\nInstalling generator-jhipster locally using yarn'));
+                    shelljs.exec('yarn install', errorLocalJhipsterInstall);
```
Comment (deepu105) [inline generators/app/index.js]: please use `spawnCommand` method from yeoman for yarn and `npmInstall` method for npm  http://yeoman.io/authoring/dependencies.html

**Comment by jdubois** [comment]
There are unit tests failing :-(
This is really an important feature for me, so big +1 , and tell me if I can help you!

**Comment by PierreBesson** [comment]
@jdubois I'm working on it. It's just that I want to be sure everything works well with npm and yarn and that it supports the additional module question.

**Comment by jdubois** [comment]
@PierreBesson do you need help on this? I should have time next week

**Comment by pascalgrimaud** [comment]
As I will rename `yarnInstall` to `clientPackageManager`, it will have impact on this PR

**Comment by PierreBesson** [comment]
I have done some changed and rebased, still to do:
[ ] respect the skipInstall flag
[ ] yarnInstall -> clientPackageManager
[ ] fix npm test, @pascalgrimaud please have a look at it, this is caused by the package.json being expected to not be present for skipClient apps.
[ ] After upgrading to Yeoman 1.0 use the yarnInstall method from yeoman rather than spawnCommand.

**Comment by pascalgrimaud** [comment]
@PierreBesson : ok, I can have a look tomorrow morning

**Comment by PierreBesson** [comment]
Sorry for taking so long. It should be OK now.

**Comment by jdubois** [comment]
Great!!!

**Pull request merged**